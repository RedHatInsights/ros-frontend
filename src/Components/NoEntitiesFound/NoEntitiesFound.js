import React from 'react';
import {
    Title,
    Button,
    EmptyState,
    EmptyStateBody,
    EmptyStateIcon,
    EmptyStateVariant,
    EmptyStatePrimary
} from '@patternfly/react-core';
import { CheckCircleIcon } from '@patternfly/react-icons';
import { GETTING_STARTED_DOC } from '../../constants';

export default function NoEntitiesFound() {
    return (
        <EmptyState variant={EmptyStateVariant.large}>
            <EmptyStateIcon icon={CheckCircleIcon} color='green'/>
            <Title headingLevel="h5" size="lg">
                No suggested instance types found
            </Title>
            <EmptyStateBody>
                {/* eslint-disable-next-line max-len */}
                All the systems that are sending data are optimized. Visit getting started documentation to add more systems to Resource Optimization to see more suggestions.
            </EmptyStateBody>
            <EmptyStatePrimary>
                <Button component='a' variant='primary' href={GETTING_STARTED_DOC} target='_blank'>Getting started documentation</Button>
            </EmptyStatePrimary>
        </EmptyState>
    );
}
