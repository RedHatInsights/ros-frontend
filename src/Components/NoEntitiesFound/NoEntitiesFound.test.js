import React from 'react';
import renderer from 'react-test-renderer';
import NoEntitiesFound from './NoEntitiesFound';

describe('NoEntitiesFound', () => {
    it('should render correctly when there are no records', () => {
        const wrapper = renderer.create(<NoEntitiesFound />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });
});
