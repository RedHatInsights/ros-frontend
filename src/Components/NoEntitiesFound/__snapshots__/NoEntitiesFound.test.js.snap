// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoEntitiesFound should render correctly when there are no records 1`] = `
<div
  className="pf-c-empty-state pf-m-lg"
>
  <div
    className="pf-c-empty-state__content"
  >
    <svg
      aria-hidden="true"
      aria-labelledby={null}
      className="pf-c-empty-state__icon"
      fill="green"
      height="1em"
      role="img"
      style={
        Object {
          "verticalAlign": "-0.125em",
        }
      }
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
      />
    </svg>
    <h5
      className="pf-c-title pf-m-lg"
      data-ouia-component-id="OUIA-Generated-Title-1"
      data-ouia-component-type="PF4/Title"
      data-ouia-safe={true}
    >
      No suggested instance types found
    </h5>
    <div
      className="pf-c-empty-state__body"
    >
      All the systems that are sending data are optimized. Visit getting started documentation to add more systems to Resource Optimization to see more suggestions.
    </div>
    <div
      className="pf-c-empty-state__primary"
    >
      <a
        aria-disabled={false}
        aria-label={null}
        className="pf-c-button pf-m-primary"
        data-ouia-component-id="OUIA-Generated-Button-primary-1"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe={true}
        disabled={null}
        href="https://access.redhat.com/documentation/en-us/red_hat_insights/2023/html/assessing_and_monitoring_rhel_resource_optimization_with_insights_for_red_hat_enterprise_linux/index"
        role={null}
        target="_blank"
        type={null}
      >
        Getting started documentation
      </a>
    </div>
  </div>
</div>
`;
